AzureStack/2015-04-01/swagger/GalleryAdmin.json

```diff
@@ -0,0 +1,502 @@
+      "get": {
+        "tags": [
+          "AdminGalleryItems"
+        ],
+        "description": "Lists gallery registrations.",
+        "operationId": "AdminGalleryItems_List",
+        "consumes": [],
+        "produces": [
+          "application/json",
+          "text/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/ResourceDefinition"
+              }
+            }
+          }
+        }
+      },
+      "post": {
+        "tags": [
+          "AdminGalleryItems"
+        ],
+        "description": "Uploads a provider gallery item to the storage.",
+        "operationId": "AdminGalleryItems_Upload",
+        "consumes": [         
+          "application/json",
+          "text/json"
+          ],
+        "produces": [
+          "application/json",
+          "text/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "galleryItemUri",
+            "in": "body",
+            "description": "Gallery Item Uri from where the gallery package is downloaded.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/GalleryItemUriPayload"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ResourceDefinition"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/ResourceDefinition"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/microsoft.gallery.admin/galleryItems/{galleryItemName}": {
+      "get": {
+        "tags": [
+          "AdminGalleryItems"
+        ],
+        "description": "Gets a specific gallery registration.",
+        "operationId": "AdminGalleryItems_Get",
+        "consumes": [],
+        "produces": [
+          "application/json",
+          "text/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "galleryItemName",
+            "in": "path",
+            "description": "Identity of the gallery item. Includes publisher name, item name and may include version separated by period character.",
```

Comment:
How is the user supposed to give that info? Can you give an example? This can be very confusing.
Please provide examples for both the scenarios: with and without version.